### PR TITLE
Add cgal/5.0

### PR DESCRIPTION
**Pull request by czoido**
Specify library name and version:  **cgal/5.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Morwenn** [comment]
According to [their documentation](https://github.com/CGAL/cgal/wiki/How-to-use-CGAL-with-CMake-or-your-own-build-system) the blessed CMake `find_package` name for the library is `CGAL`, so it would probably be a good idea to set `cpp_info.names` accordingly for the `cmake_find_package` and `cmake_find_package_multi` generators.

File: recipes/cgal/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            "project(CGAL CXX C)", '''project(CGAL CXX C)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()''')
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "CGAL-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+        self._patch_sources()
```
Comment (uilianries) [inline recipes/cgal/all/conanfile.py]:
```suggestion
```

File: recipes/cgal/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    def build(self):
```
Comment (uilianries) [inline recipes/cgal/all/conanfile.py]:
```suggestion
    def build(self):
        self._patch_sources()
```

File: recipes/cgal/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE*", dst="licenses", src=os.path.join(self._source_subfolder))
```
Comment (uilianries) [inline recipes/cgal/all/conanfile.py]:
```suggestion
        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
```

**Pull request merged**